### PR TITLE
fix: getHealth returns 206 status code if EL is syncing / optimistic

### PR DESCRIPTION
**Pull request by AkshatGada**
Solves #6664

### CODE REVIEW

File: packages/validator/src/validator.ts [hunk 1]
```diff
@@ -136,7 +136,10 @@ export class Validator {
 
         this.clock.runEverySlot(() =>
           this.fetchBeaconHealth()
-            .then((health) => metrics.beaconHealth.set(health))
+            .then((health) => {metrics.beaconHealth.set(health);
+                   if(health == 0){ 
+                    this.logger.warn("Node is unhealthy");
+                   }})
```
Comment (nflaig) [inline packages/validator/src/validator.ts:142]: unrelated changes

```suggestion
            .then((health) => metrics.beaconHealth.set(health))
```

File: packages/beacon-node/src/api/impl/node/index.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ export function getNodeApi(
 
       let healthStatus: number;
 
-      if (sync.getSyncStatus().isSyncing) {
+      if (sync.getSyncStatus().isSyncing ||  (!sync.getSyncStatus().isSyncing  && syncingStatus.getSyncStatus().isOptimistic)) {
```
Comment (nflaig) [inline packages/beacon-node/src/api/impl/node/index.ts]: should avoid calling `.getSyncStatus()` multiple times, the condition is also either node is syncing **or** EL is optimistic

see https://github.com/ethereum/beacon-APIs/pull/442/files#diff-cc32cc26efdb358fa3338a84d59b700df00098f61d1ecbdacd5545ed0bccb31aR20

**Comment by nflaig** [comment]
> [CLA assistant check](https://cla-assistant.io/ChainSafe/lodestar?pullRequest=6837) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ChainSafe/lodestar?pullRequest=6837) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ChainSafe/lodestar?pullRequest=6837) it.

could you sign this to make CI happy please

File: packages/api/src/beacon/routes/node.ts [hunk 3]
```diff
@@ -175,14 +175,33 @@ export function getReqSerializers(): ReqSerializers<Api, ReqTypes> {
...
+     * // The `response.body` will be the response body.
+/**
+     * Define the request serializers for the `getHealth` route.
+     *
+     * @returns {ReqSerializers<Api, ReqTypes>["getHealth"]} The request serializers for the `getHealth` route.
+     *
+     * @example
+     * // Send a GET request to the "/eth/v1/node/health" endpoint with the `syncing_status` query parameter set to 10.
+     * const response = await client.getHealth({syncingStatus: 10});
+     * // The `response.body` will be the response body.
```
Comment (nflaig) [inline packages/api/src/beacon/routes/node.ts:195]: What is this comment?
```
The `response.body` will be the response body.
```

I doubt you manually added this, was it AI generated?

File: packages/beacon-node/src/api/impl/node/index.ts [hunk 4]
```diff
@@ -75,7 +75,7 @@ export function getNodeApi(
 
       let healthStatus: number;
 
-      if (sync.getSyncStatus().isSyncing) {
+      if (sync.getSyncStatus().isSyncing ||  syncingStatus.getSyncStatus().isOptimistic) {
```
Comment (nflaig) [inline packages/beacon-node/src/api/impl/node/index.ts:78]: this still calls `.getSyncStatus()` twice if `isSyncing === false`, we should avoid that

File: packages/api/src/beacon/routes/node.ts (hunk 3 above)
Comment (Giri-Aayush) [reply packages/api/src/beacon/routes/node.ts:195]: I believe that is the case.

**Comment by nflaig** [comment]
Closed in favor of https://github.com/ChainSafe/lodestar/pull/6920
